client/app_control.cpp

```diff
@@ -149,6 +149,36 @@ bool ACTIVE_TASK_SET::poll() {
             }
         }
     }
+
+    // check if a job is "stuck" (did not make progress in the last hour)
+    // notify the user about the issue
+    // abort after some time
+    static double last_stuck_check_time = 0;
+    if (gstate.now - last_stuck_check_time > STUCK_CHECK_POLL_PERIOD) {
+        for (i=0; i<active_tasks.size(); i++){
+            ACTIVE_TASK* atp = active_tasks[i];
+            if (!atp->non_cpu_intensive()) continue;
```

Comment:
Correct me if I'm wrong but this check will skip all the normal tasks, and will iterate through Non-CPU intensive tasks only.
Is this a typo and `!` is not needed here?

Reply:
Ah i believe that is correct, that `!` shouldn't be there